src/node/biquad_filter.rs

```diff
@@ -0,0 +1,1180 @@
+            " InvalidAccessError: All paramaters should be the same length"
+        );
+        assert_eq!(
+            mag_response.len(),
+            phase_response.len(),
+            " InvalidAccessError: All paramaters should be the same length"
+        );
+
+        // Ensures that given frequencies are in the correct range
+        let min = 0.;
+        let max = self.sample_rate / 2.;
+        for f in frequency_hz.iter_mut() {
+            *f = f.clamp(min, max);
+        }
+    }
+
+    /// Mock of `get_frequency_response`
+    /// This function is the same as `get_frequency_response` except it never send the `CoeffsReq`.
+    /// In tests, we use `OfflineAudioContext` and in this context the `CoeffsReq` is not sendable.
+    ///
+    /// # Arguments
+    ///
+    /// * `frequency_hz` - frequencies for which frequency response of the filter should be calculated
+    /// * `mag_response` - magnitude of the frequency response of the filter
+    /// * `phase_response` - phase of the frequency response of the filter
+    #[cfg(test)]
+    fn get_frequency_response_mock(
+        &self,
+        frequency_hz: &mut [f32],
+        mag_response: &mut [f32],
+        phase_response: &mut [f32],
+    ) {
+        self.validate_inputs(frequency_hz, mag_response, phase_response);
+    }
+}
+
+#[derive(Debug)]
+struct Params {
+    q: f32,
+    detune: f32,
+    frequency: f32,
+    gain: f32,
+    type_: BiquadFilterType,
+}
+
+struct RendererConfig {
+    sample_rate: f32,
+    q: AudioParamId,
+    detune: AudioParamId,
+    frequency: AudioParamId,
+    gain: AudioParamId,
+    type_: Arc<AtomicU32>,
+    params: Params,
+    receiver: Receiver<CoeffsReq>,
+}
+
+/// Biquad filter coefficients
+#[derive(Clone, Copy, Debug)]
+struct Coefficients {
+    // Denominator coefficients
+    a1: f64,
+    a2: f64,
+
+    // Nominator coefficients
+    b0: f64,
+    b1: f64,
+    b2: f64,
+}
+
+struct BiquadFilterRenderer {
+    sample_rate: f32,
+    q: AudioParamId,
+    detune: AudioParamId,
+    frequency: AudioParamId,
+    gain: AudioParamId,
+    type_: Arc<AtomicU32>,
+    ss1: [f64; MAX_CHANNELS],
+    ss2: [f64; MAX_CHANNELS],
+    coeffs: Coefficients,
+    receiver: Receiver<CoeffsReq>,
+}
+
+impl AudioProcessor for BiquadFilterRenderer {
+    fn process(
+        &mut self,
+        inputs: &[crate::alloc::AudioBuffer],
+        outputs: &mut [crate::alloc::AudioBuffer],
+        params: AudioParamValues,
+        _timestamp: f64,
+        _sample_rate: SampleRate,
+    ) {
+        // single input/output node
+        let input = &inputs[0];
+        let output = &mut outputs[0];
+
+        let g_values = params.get(&self.gain);
+        let det_values = params.get(&self.detune);
+        let freq_values = params.get(&self.frequency);
+        let q_values = params.get(&self.q);
+
+        self.filter(input, output, g_values, det_values, freq_values, q_values);
+    }
+
+    fn tail_time(&self) -> bool {
+        false
```

Comment:
Todo: true! Spec has some recommendations

Reply:
That's fine for now, but it has a performance penalty.
Let's make a separate issue to handle tail time for the biquad and iirfilternode

Reply:
issue #34